### DISCUSSION TITLE
Aliasing selected field on prisma so it doesn't return object?

### DISCUSSION THREAD

**Discussion by mandaputtra** [prisma/prisma#5428] [Q&A] [answered]
How to alias the field that you selected on prisma? Suppose I have this query

```js
      const orders = await this.prisma.order.findMany({
        where,
        orderBy: { [sortBy]: orderAs },
        include: { product: true }
      });
```

But it will be returning a product object

```
{
  id: 1
  ....
  product: { .... .... }
}
```

I want something like these

```
{
  id: 1
  .....
  productName: ....
  productCode: ....
}
```

How do I do that in prisma way? Or should I use raw query instead?

<details>
<summary>schema.prisma</summary>

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

// -- START ENUM DECLARATION --
enum ProductCategory {
  EWALLET
  PULSA
  GOODS
}

enum OrderStatus {
  PENDING
  SUCCESS
  FAILED
  PROCESSED
}

// -- END ENUM DECLARATION --

model Vendor {
  id        Int       @id @default(autoincrement())
  name      String    @unique
  email     String    @unique
  remarks   String
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  products  Product[]
}

model Product {
  id        Int             @id @default(autoincrement())
  vendor    Vendor          @relation(fields: [vendorId], references: [id])
  vendorId  Int
  name      String
  code      String
  price     Int
  category  ProductCategory @default(EWALLET)
  createdAt DateTime        @default(now())
  updatedAt DateTime        @updatedAt
  orders    Order[]
}

model Order {
  id            Int         @id @default(autoincrement())
  product       Product     @relation(fields: [productId], references: [id])
  productId     Int
  remarks       String?
  status        OrderStatus @default(PROCESSED)
  customerNo    String?
  customerEmail String?
  fee           Int
  price         Int
  refId         String?
  index         Int         @default(0)
  bulkId        String?
  createdAt     DateTime    @default(now())
  updatedAt     DateTime    @updatedAt
}

```

</details>

**Comment by ryands17** [comment, marked as answer]
@mandaputtra 👋 

What you want here is to flatten the object which is currently not supported by Prisma. It will return data in the exact format that your relations are, so you would need to customize this in your application logic itself. As for aliasing fields, you can follow [this](https://github.com/prisma/prisma/issues/4433) feature request for more info.

**Reply by mandaputtra** [reply]
Thanks a lot

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
